src/dotnet-sourcelink/Program.cs

```diff
@@ -176,20 +178,14 @@ private static async Task<int> TestAsync(
             string? user,
             string? password,
             bool offline,
-            ParseResult parseResult)
+            ParseResult parseResult,
+            CancellationToken cancellationToken)
         {
             var authenticationHeader = (authMethod != null) ? GetAuthenticationHeader(authMethod, authEncoding ?? Encoding.ASCII, user!, password!) : null;
 
-            var cancellationSource = new CancellationTokenSource();
-            Console.CancelKeyPress += (sender, e) =>
```

Comment:
this is not required, S.CL will do this on it's own, we just need to pass the `CancellationToken`

https://github.com/dotnet/command-line-api/blob/060374e56c1b2e741b6525ca8417006efb54fbd7/src/System.CommandLine/Invocation/ProcessTerminationHandler.cs#L45-L46